### PR TITLE
provider/aws: Limit the number of retry attempts when deleting a Subnet

### PR DESCRIPTION
**Pull request by catsby**
This PR is a compromise to try and satisfy both #934 and the issue raised in #1368.

TL;DR: 

- retry the delete call if it's a dependency violation, but no more than 10 attempts. 
- shorten overall timeout period 

In practice, I was able to fully delete the config provided in #934 without error. 
I am also receiving the necessary dependency violation error message under 1 minute in the situation described in #1368, where an external dependency unknown to Terraform is introduced, in which case the dependency violation is never resolved. 

Unfortunately we can't simply rely on a shorter timeout, because once the timeout is hit, we receive a timeout error, and the underlying error is lost. 

If this is still unacceptable, we'll likely apply #1368 and instruct users who hit the dependency violations to simply retry, or add explicit `depends_on` relationships in their configuration.

### CODE REVIEW

**Comment by mitchellh** [comment]
LGTM. Sad, but makes sense.

**Comment by bitglue** [comment]
You can't polish a turd, as they say. A minute is still a long time to wait for something that's guaranteed to never succeed. It's also not long enough to wait to be guaranteed to fix the problem as you intend. For example, RDS can take a lot more than a minute to delete.

#934 is easily solved without kludges by reading [the documentation](https://www.terraform.io/intro/getting-started/dependencies.html). How about pointing the user to that whenever there is a DependencyViolation? There are only two reasons that error would be encountered:

  - Terraform would have to delete a resource it did not create in order to proceed (correct behavior: exit immediately with an error)
  - The configuration is wrong (correct behavior: make it clear to the user how to fix it)
